activities.el

```diff
@@ -479,12 +479,47 @@ It will not be recoverable."
   "Automatically save activities when Emacs has been idle this many seconds."
   :type 'natnum)
 
+(defvar activities-mode-map (make-sparse-keymap)
+  "The mode keymap for `activities-mode'.")
+
+(defun activities-mode-line-format ()
+  (when-let ((cur (activities-current)))
+    (let ((cur-activity-title (concat " " (activities-name-for cur))))
+      `(:propertize
+        ,cur-activity-title
+        mouse-face mode-line-highlight
+        help-echo
+        ,(lambda (&rest _)
+           (concat
+            (format "Current activity:%s\n" cur-activity-title)
+            "mouse-1: Display minor mode menu\n"
+            "mouse-2: Show help for minor mode"))
+        keymap
+        ,(let ((map (make-sparse-keymap)))
+           (define-key map [mode-line down-mouse-1]
+                       activities-mode-line-menu)
+           (define-key map [mode-line down-mouse-3]
+                       activities-mode-line-menu)
+           (define-key map [mode-line mouse-2]
+                       (lambda ()
+                         (interactive)
+                         (describe-function 'activities-mode)))
+           map)))))
```

Comment:
1. Nitpick, but I prefer not to abbreviate variable names in most cases, e.g. `cur`.
2. Probably the keymap should be defined statically.
3. I guess the lambda could be defined statically as well, i.e. not be a closure over the title variable, because it should be able to get the current activity name by doing what this function does when necessary.
4. The function should have a docstring.  :)

Reply:
Done. I left the closure and keymap as is since that seems to be the general approach via informal surveillance (and it works fine and without any overall performance issues).